### PR TITLE
SSS: Call graph utilities

### PR DESCRIPTION
**Pull request by smowton**
These extend `call_grapht` with some basic is-(transitive)-child and topological sorting functions.

### CODE REVIEW

File: src/analyses/call_graph.cpp [hunk 1]
```diff
@@ -121,6 +121,27 @@ void call_grapht::output_dot(std::ostream &out) const
   out << "}\n";
 }
 
+
+void call_grapht::output_dot(
+  goto_functionst const &functions,
```
Comment (peterschrammel) [inline src/analyses/call_graph.cpp]: const qualifier first (whole file)

File: src/analyses/call_graph.cpp [hunk 2]
```diff
@@ -121,6 +121,27 @@ void call_grapht::output_dot(std::ostream &out) const
   out << "}\n";
 }
 
+
```
Comment (peterschrammel) [inline src/analyses/call_graph.cpp]: function comment blocks

**Comment by smowton** [comment]
done

File: src/analyses/call_graph.cpp (hunk 1 above)
Comment (peterschrammel) [inline src/analyses/call_graph.cpp]: done?

Comment (smowton) [inline src/analyses/call_graph.cpp]: Yes, this shows up an <outdated> block for me-- or see the Files tab for confirmation

Comment (peterschrammel) [inline src/analyses/call_graph.cpp]: Indeed, everything except https://github.com/diffblue/cbmc/pull/597/files#diff-6de119d510b9252c9cd42ded5c173b12R122 onwards

Comment (smowton) [inline src/analyses/call_graph.cpp]: Oops, really done this time.

**Pull request merged**